File: backend/src/main/java/com/example/backend/service/StudentService.java
package com.example.backend.service;

import com.example.backend.entity.Course;
import com.example.backend.entity.Student;

import java.util.Iterator;
import java.util.List;

public interface StudentService {

    void create(Student student);
    Student getById(Long id);

    List<Student> getAll();

    List<Student> getStudentsByMajor(String major);

    void update(Student student);

    void delete(Long id);

    List<Course> getCoursesByStudentId(Long id);
}

File: front-end/src/services/StudentService.ts
import fetch from "./FetchInspecor";

const studentService = {

    getAll: async (major: string) => {
        return await fetch.get(`/students?major=${major}`)
    },

    delete: async (id: number) => {
        await fetch.delete(`/students/${id}`)
    }

    ,

    create: async (student: any) => {
        await fetch.post(`/students`, student)
    },


    update: async (id: number, student: any) => {
        await fetch.put(`/students/${id}`, student)
    }
}


export default studentService;